java/operator/src/main/java/com/instaclustr/cassandra/operator/k8s/K8sResourceUtils.java

```diff
@@ -35,98 +38,121 @@ public K8sResourceUtils(final ApiClient apiClient,
-                () -> apiClient.execute(replaceResourceCall)
-        );
-    }
-
-    public void deleteResource(final Call deleteResourceCall) throws ApiException {
-        apiClient.execute(deleteResourceCall);
-    }
-
-    public void createOrReplaceNamespaceService(final V1Service service) throws ApiException {
-        createOrReplaceResource(
-                coreApi.createNamespacedServiceCall(service.getMetadata().getNamespace(), service, null, null, null),
-                //coreApi.replaceNamespacedServiceCall(service.getMetadata().getName(), service.getMetadata().getNamespace(), service, null, null, null)
-                // temporarily disable service replace call to fix issue #41 since service can't be customized right now
-                coreApi.readNamespacedServiceCall(service.getMetadata().getName(), service.getMetadata().getNamespace(), null, null, null, null, null)
-        );
+    public void createOrReplaceNamespacedService(final V1Service service) throws ApiException {
+        try (@SuppressWarnings("unused") final MDC.MDCCloseable _serviceMDC = putNamespacedName("Service", service.getMetadata())) {
+            final String namespace = service.getMetadata().getNamespace();
+
+            logger.debug("Creating/replacing namespaced Service.");
+
+            createOrReplaceResource(
+                    () -> {
+                        coreApi.createNamespacedService(namespace, service, null, null, null);
+                        logger.debug("Created namespaced Service.");
+                    },
+                    //coreApi.replaceNamespacedService(service.getMetadata().getName(), service.getMetadata().getNamespace(), service, null)
+                    // temporarily disable service replace call to fix issue #41 since service can't be customized right now
+                    () -> {}
+            );
+        }
     }
 
-    public void createOrReplaceNamespaceConfigMap(final V1ConfigMap configMap) throws ApiException {
-        final String namespace = configMap.getMetadata().getNamespace();
-
-        createOrReplaceResource(
-                coreApi.createNamespacedConfigMapCall(namespace, configMap, null, null, null),
-                coreApi.replaceNamespacedConfigMapCall(configMap.getMetadata().getName(), namespace, configMap, null, null, null)
-        );
+    public void createOrReplaceNamespacedConfigMap(final V1ConfigMap configMap) throws ApiException {
+        try (@SuppressWarnings("unused") final MDC.MDCCloseable _configMapMDC = putNamespacedName("ConfigMap", configMap.getMetadata())) {
+            final String namespace = configMap.getMetadata().getNamespace();
+
+            logger.debug("Creating/replacing namespaced ConfigMap.");
+
+            createOrReplaceResource(
+                    () -> {
+                        coreApi.createNamespacedConfigMap(namespace, configMap, null, null, null);
+                        logger.debug("Created namespaced ConfigMap.");
```

Comment:
we should log the namespace/name of the resource created so we can look it up via kubectl if needed (Applies to all create or replace resource invocations)

Reply:
We do (subtly). The resource namespace & name is added to the Message Diagnostic Context (MDC) in the try-with-resources block. The operator logback message format contains the `%mdc` token, which will include the contents of the MDC with every message.